selection_tools/prepare_histograms.C

```diff
@@ -50,25 +51,52 @@ TH1D* fill_and_save_histograms(const vector<Event*> &events, string output_path)
       histogramFiller.fill_hists(muon_1, muon_2, event, "os");
       histogramFiller.fill_final_selection_hists(muon_1, muon_2, event, "os");
 
-      if(!muon_1 || !muon_2) continue;
-      if(!muon_1->has_alp_ancestor(event->particles)) continue;
-      if(!muon_2->has_alp_ancestor(event->particles)) continue;
-      auto mother_1 = event->particles[muon_1->mothers[0]];
-      auto mother_2 = event->particles[muon_2->mothers[0]];
-      if(mother_1 != mother_2) continue;
-      histogramFiller.fill_alp_hists(muon_1, muon_2, mother_1, event, "os");
+      if(!reduce_hists)
```

Comment:
A small trick: if you can move this block to the end of the method, then you can invert the logic and do `if(reduce_hists) return;`. This has two advantages: you decrease nested levels (easier to read) and also when someone who reads the code set `reduce_hists = true`, then they can immediately stop reading what's after this line.